server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailSubmissionSetMethod.scala

```diff
@@ -250,30 +255,71 @@ class EmailSubmissionSetMethod @Inject()(serializer: EmailSubmissionSetSerialize
 
   private def sendEmail(mailboxSession: MailboxSession,
                         request: EmailSubmissionCreationRequest): SMono[(EmailSubmissionCreationResponse, MessageId)] =
-   for {
+    for {
       message <- SFlux(messageIdManager.getMessagesReactive(List(request.emailId).asJava, FetchGroup.FULL_CONTENT, mailboxSession))
         .next
         .switchIfEmpty(SMono.error(MessageNotFoundException(request.emailId)))
       submissionId = EmailSubmissionId.generate
       message <- SMono.fromTry(toMimeMessage(submissionId.value, message))
       envelope <- SMono.fromTry(resolveEnvelope(message, request.envelope))
       _ <- validate(mailboxSession)(message, envelope)
+      parameters = envelope.mailFrom.parameters
+      _ <- SMono.fromTry(validateFromParameters(parameters))
+      delay <- SMono.fromTry(retrieveDelay(parameters))
+      _ <- SMono.fromTry(validateDelay(delay))
+
       mail = {
         val mailImpl = MailImpl.builder()
           .name(submissionId.value)
-          .addRecipients(envelope.rcptTo.map(_.email).asJava)
+          .addRecipients(envelope.rcptTo.filter(m => m.parameters.isEmpty || !(m.parameters.get.contains(ParameterName.holdFor) || m.parameters.get.contains(ParameterName.holdUntil))).map(_.email).asJava)
           .sender(envelope.mailFrom.email)
           .addAttribute(new Attribute(MAIL_METADATA_USERNAME_ATTRIBUTE, AttributeValue.of(mailboxSession.getUser.asString())))
           .build()
         mailImpl.setMessageNoCopy(message)
         mailImpl
       }
-      _ <- SMono(queue.enqueueReactive(mail))
+
+      _ <- SMono(queue.enqueueReactive(mail, delay))
         .`then`(SMono.fromCallable(() => LifecycleUtil.dispose(mail)).subscribeOn(Schedulers.boundedElastic()))
         .`then`(SMono.just(submissionId))
+      sendAt = UTCDate(ZonedDateTime.now(clock).plus(delay))
     } yield {
-      EmailSubmissionCreationResponse(submissionId) -> request.emailId
+      EmailSubmissionCreationResponse(submissionId, sendAt) -> request.emailId
+    }
+
+  private def retrieveDelay(mailParameters: Option[Map[ParameterName, Option[ParameterValue]]]): Try[Duration] = mailParameters match {
+      case None => Success(NO_DELAY)
+      case Some(aMap) if aMap.contains(ParameterName.holdFor) && aMap(ParameterName.holdFor).nonEmpty => Try(Duration.ofSeconds(aMap(ParameterName.holdFor).get.value.toLong))
+      case Some(aMap) if aMap.contains(ParameterName.holdFor) && aMap(ParameterName.holdFor).isEmpty => Success(Duration.ZERO)
```

Comment:
Good candidate to a refactoring: the option case can be handled without pattern matching